HIP/hip-873.md

```diff
@@ -0,0 +1,103 @@
+---
+hip: 873
+title: Adding Decimal Information to Token Balance Calls in the Hedera REST API
+author: Houdini NFT (@HoudiniNFT), Ty Smith (@ty-swirldslabs)
+requested-by: Atlantis
+type: Standards Track
+category: Mirror
+needs-council-approval: Yes
+status: Review
+created: 2024-01-30
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/874
+updated: 2024-01-31
+---
+
+### Abstract
+This HIP proposes an enhancement (with backwards compatibility) to the Hedera REST API to include decimal information in the token balance queries. Currently, users face challenges in accurately determining token balances due to the absence of decimal details in the `/tokens` and `/account/{id}/tokens` endpoints. This proposal aims to reduce the number of API calls needed and streamline the process of retrieving comprehensive token information.
+
+### Motivation
+Retrieving token balance information without decimal details requires users to make multiple API calls, leading to inefficiencies and increased computational load. Decimals should be supplied with the balance amount. Specifically, users must perform additional queries to obtain the decimal count for each token associated with an account. This is crucial for determining the correct token balance. This process is cumbersome, time-consuming, and can lead to increased costs and latency in applications relying on the Hedera API.
+
+### Specification
+The proposed solution involves modifying the response structure of two endpoints:
+
+**Route:**  
+`/api/v1/tokens?account.id={account id}`
+
+**Current Response Structure:**
+```json
+{
+  "tokens": [
+    {
+      "admin_key": null,
+      "symbol": "PEC_E",
+      "token_id": "0.0.732556",
+      "type": "NON_FUNGIBLE_UNIQUE"
+    }
+  ]
+}
+```
+
+**Proposed Response Structure:**
+```json
+{
+  "tokens": [
+    {
+      "admin_key": null,
+      "symbol": "PEC_E",
+      "name": "[Token Name]",
+      "token_id": "0.0.732556",
+      "type": "NON_FUNGIBLE_UNIQUE",
+      "decimals": [Decimals]
+    }
+  ]
+}
+```
+
+**Route:**  
+`/api/v1/accounts/{account id}`
+
+**Current Response Structure:**
+```json
+{
+  "tokens": [
+    {
+      "token_id": "0.0.732556",
+      "balance": 0
```

Comment:
There's other fields currently in this response that are not shown here. We should add them here to reduce confusion or add a note that some fields are omitted for brevity.

Reply:
Added fields not shown